tests/cppException/enc/cppException.cpp

```diff
@@ -247,7 +247,7 @@ bool BasicVerification()
     {
         return false;
     }
-    catch (BarClass02 ex_obj)
+    catch (BarClass02& ex_obj)
```

Comment:
I think this was intentionally a catch by value based on the test comment, since it's looking for an exact class match.

Conversely, it seems like `BarClass01` should also be caught by reference for lines 283, 321 and 424 (for consistency with the updated catch on line 412), even though it doesn't matter for an exception class with no inheritance. i.e. the tests seem to checking for the catching of the correct type rather than mechanism of catching (as in lines 162/194) and catching consistently by reference eliminates an extra variable.

Reply:
Updated PR to address this but squashed into the same commit.